zcash_primitives/src/transaction/components.rs

```diff
@@ -1,603 +1,19 @@
 //! Structs representing the components within Zcash transactions.
 
-use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
-
-use ff::PrimeField;
-use group::GroupEncoding;
-
-use std::io::{self, Read, Write};
-
-#[cfg(feature = "zfuture")]
-use std::convert::TryFrom;
-
-use crate::{
-    legacy::Script,
-    primitives::Nullifier,
-    redjubjub::{PublicKey, Signature},
+pub mod amount;
+pub mod sapling;
+pub mod sprout;
+pub mod transparent;
+pub mod tze;
```

Comment:
These being public adds new surface to the public API, but because we then `pub use` the values, there's no change from before.